db/tenants.go

```diff
@@ -228,8 +228,79 @@ func (db *DB) GetTenantRole(org string, team string) (uuid.UUID, string, error)
 	return uuid.Parse(tenantUUID), role, nil
 }
 
-func (db *DB) DeleteTenant(tenant *Tenant) error {
+func (db *DB) DeleteTenant(tenant *Tenant, recurse bool) error {
+	//delete tasks
+	if recurse {
+		err := db.Exec(`
+			DELETE FROM tasks
+			WHERE stopped_at IS NOT NULL 
+			AND tenant_uuid = ?`, tenant.UUID.String())
+		if err != nil {
+			return fmt.Errorf("unable to delete tenant tasks: %s", err)
+		}
+
+		err = db.Exec(`
+			DELETE FROM memberships 
+			WHERE tenant_uuid = ?`, tenant.UUID.String())
+		if err != nil {
+			return fmt.Errorf("unable to delete tenant memberships: %s", err)
+		}
+
+		//delete jobs
+		err = db.Exec(`
+			DELETE FROM jobs
+			WHERE tenant_uuid = ?`, tenant.UUID.String())
+		if err != nil {
+			return fmt.Errorf("unable to delete tenant jobs: %s", err)
+		}
+
+		err = db.Exec(`
+			UPDATE targets
+			SET tenant_uuid = ''
+			WHERE tenant_uuid = ?`, tenant.UUID.String())
+		if err != nil {
+			return fmt.Errorf("unable to clear tenant targets: %s", err)
+		}
+
+		err = db.Exec(`
+			UPDATE stores
+			SET tenant_uuid = ''
+			WHERE tenant_uuid = ?`, tenant.UUID.String())
+		if err != nil {
+			return fmt.Errorf("unable to clear tenant stores: %s", err)
```

Comment:
not really clearing them so much as marking them for deletion

Reply:
From the point-of-view of the caller, they are gone.  Saying that we are 'marking them for deletion so that a future background process can eventually clean them up, but you won't be able to see or recover them once this has been kicked off' isn't as helpful.